### ISSUE TITLE
Allow a user to create new API tokens from their 'my account' page

### ISSUE THREAD

**Issue by prioux** [aces/cbrain#965]
A simple button would let a user generate a new API token, with associated information: this token becomes invalid after three days of not being used, and can only be used from the same IP address as its first use.

Implementation: duplicate the CbrainLargeSession object, clean it up of filters, adjust IP address and hostnames in it. Also: flag it as a special session somehow, and make sure the IP invalidation code doesn't get triggered the first time.

**Comment by prioux** [comment]
Also: only one token valid at a time, and token disappears from page forever after first view.

Could be useful for LORIS and NeuroHub integrations.

**Comment by prioux** [comment]
Implementation notes:

```ruby
# Extract the LargeSessionInfo object for the current user's session
lif = current_session.instance_eval { @model } # maybe add a method to CbrainSession class?

# Prep it for a new token
newl_lif = lif.dup
new_lif.session_id = SecureRandom.hex

# Encapsulate it temporarily so we can clean interactive session info (scopes etc)
fake_cbsess = CbrainSession.new(new_lif)
fake_cbsess.mark_as_modified
fake_cbsess.clear
fake_cbsess.clean_data_for_api

# Save the object in the DB
fake_cbsess.touch_unless_recent # forces save in DB of new_lif
# OR ... ?
new_lif.save

# Show token in page: new_lif.session_id
new_lif.session_id  # basically the value from SecureRandom
```

**Comment by prioux** [comment]
I'd wrap the code above in the CbrainSession class as a new instance method 'dup_with_new_token' so that in the controller we now simply have

```ruby
new_sess                = cbrain_session.dup_with_new_token
@new_temp_token = new_sess.cbrain_api_token
```

**Comment by prioux** [comment]
I'm adding the NeuroHub label because it could be useful for the LORIS integration. Also for some stuff that other LORIS groups are working on that I know about (shout out to @laemtl)

**Comment by prioux** [comment]
The MyAccount page referred in this issue's description is the CBRAIN page, not the NeuroHub page.

**Comment by prioux** [comment]
Implemented in #995
